setuptools/dist.py

```diff
@@ -731,13 +731,13 @@ def finalize_options(self):
         to influence the order of execution. Smaller numbers
         go first and the default is 0.
         """
-        hook_key = 'setuptools.finalize_distribution_options'
+        group = 'setuptools.finalize_distribution_options'
 
         def by_order(hook):
             return getattr(hook, 'order', 0)
-        eps = pkg_resources.iter_entry_points(hook_key)
+        eps = map(lambda e: e.load(), pkg_resources.iter_entry_points(group))
```

Comment:
One could use `operator.methodcaller(group)` instead of the lambda, but that was introduced in python 2.6 and I was worried about that and the extra import.

Reply:
just use a list comprehension and eps.sort(key=by_order) ?

Reply:
I don't like the double iteration end extra sorting step that would lead to:
```
eps = [e.load() for e in iter_entry_points(group)]
eps.sort(key=by_order)
for ep in eps:
    ep(self)
```
Note the two `for`.

Reply:
sorted returns a materialized list, so from my point of view its not that different in terms of algorithmic complexity
please keep in mind that  `map(lambda ...:..., foo)` is just like `(... for . in foo)`

Reply:
Meh, I don't really have a horse in this race. I'll do whatever the majority decides.

Btw. we could have had our conversation in Cafe Central, I'm also currently in Innsbruck :smile:

Reply:
Happy to run with this as-is. We may choose to iterate on it after. Thanks Ronny for the feedback.